src/app/beer_garden/config.py

```diff
@@ -1261,6 +1261,11 @@ def _parse_args(args: Sequence[str]) -> Tuple[YapconfSpec, dict]:
                 },
             },
         },
+        "allow_update_systems": {
```

Comment:
I would call this `allow_system_command_updates`, since it's really just the commands that are prevented from changing.  There are other properties of a system that can be updated regardless of this setting.

Also, I would add a `long_description` here. Something like:
> When False, this prevents changes to the command definitions of a registered version of a system. This means that the system will fail to start if the commands do not match what is on record for that version of the system.  When True, the system will be allowed to start and the commands on record will be updated accordingly. NOTE: System versions containing "dev" are exempted from this check.

Reply:
That's a much better config name. I tried to keep it succinct to fit with the rest of the config, but really it just makes its use less obvious.